dfsclient/src/main/java/org/hpccsystems/dfs/client/RowServiceInputStream.java

```diff
@@ -614,8 +614,12 @@ private void readDataInFetch()
             int bytesToRead = 0;
             try
             {
+                // Read at least 512 bytes at a time
+                bytesToRead = Math.max(512,this.dis.available());
```

Comment:
what's the significance of 512?

Reply:
It was a value I found through benchmarking. I will add a constant for it and add a comment explaining it usage